checker/src/org/checkerframework/checker/unsignedness/UnsignednessAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,160 @@
+package org.checkerframework.checker.unsignedness;
+
+import org.checkerframework.checker.unsignedness.qual.*;
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.qual.TypeUseLocation;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.treeannotator.ListTreeAnnotator;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.framework.util.defaults.QualifierDefaults;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import com.sun.source.tree.BinaryTree;
+import com.sun.source.tree.CompoundAssignmentTree;
+import com.sun.source.tree.Tree;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.type.TypeMirror;
+
+/**
+ * UnsignednessAnnotatedTypeFactory removes implicit types from local variables
+ * because currently types are assigned implicitely rather than defaultly because
+ * they are linked to types, not locations. This ensures users may add Unsigned
+ * types to local variables.
+ *
+ * Furthermore, this removes all annotations from booleans to stop Unsigned types
+ * from bubbling up through comparisons.
+ *
+ * @checker_framework.manual #unsignedness-checker Unsignedness Checker
+ */
+public class UnsignednessAnnotatedTypeFactory extends BaseAnnotatedTypeFactory{
+
+    private final AnnotationMirror UNSIGNED;
+    private final AnnotationMirror SIGNED;
+    private final AnnotationMirror UNKNOWN_SIGNEDNESS;
+
+    // These are commented out until issues with making boxed implicitely signed
+    // are worked out.
```

Comment:
Before this is merged, an issue needs to be opened explaining what the problems are with boxed primitives, and the issue should be referenced here.